pandas/io/formats/csvs.py

```diff
@@ -33,27 +35,26 @@ class CSVFormatter:
     def __init__(
         self,
         obj,
-        path_or_buf=None,
-        sep=",",
-        na_rep="",
+        path_or_buf: Optional[FilePathOrBuffer[str]] = None,
+        sep: str = ",",
+        na_rep: str = "",
         float_format=None,
         cols=None,
-        header=True,
-        index=True,
+        header: Union[bool, Sequence[str]] = True,
+        index: bool = True,
         index_label=None,
-        mode="w",
-        encoding=None,
+        mode: str = "w",
+        encoding: Optional[str] = None,
         compression="infer",
-        quoting=None,
-        line_terminator="\n",
-        chunksize=None,
-        quotechar='"',
-        date_format=None,
-        doublequote=True,
+        quoting: Optional[int] = None,
+        line_terminator: Optional[str] = "\n",
+        chunksize: Optional[int] = None,
+        quotechar: Optional[str] = '"',
```

Comment:
```suggestion
        quotechar: str = '"',
```

I don't think this is optional

Reply:
Keep it as 

```python
quotechar: Optional[str] = '"'
```
?

---

mypy's log:

```
$ mypy pandas
pandas/io/formats/csvs.py:88: error: Incompatible types in assignment (expression has type "None", variable has type "str")
Found 1 error in 1 file (checked 816 source files)
```

Reply:
I think similar comment as line_terminator